static/src/javascripts/projects/common/modules/identity/upsell/page/ConfirmEmailPage.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+import React from 'preact-compat';
+
+import { NewsLetterSignUps } from './NewsLetterSignUps';
+import { OptOuts } from './OptOuts';
+import { Header } from '../header/Header';
+import { AccountCreationBlock } from '../account-creation/AccountCreationBlock';
+
+type Props = {
+    csrfToken: string,
+    accountToken: ?string,
+    email: string,
+    hasPassword: boolean,
+    hasSocialLinks: boolean,
+    isUserLoggedIn: boolean,
+};
+
+export const ConfirmEmailPage = (props: Props) => {
+    let components = [<NewsLetterSignUps />, <OptOuts />];
+
+    if (!props.hasPassword && !props.hasSocialLinks && props.accountToken) {
+        components = [
+            <AccountCreationBlock
+                csrfToken={props.csrfToken}
+                accountToken={props.accountToken}
+                email={props.email}
+            />,
+            <NewsLetterSignUps />,
+            <OptOuts />,
+        ];
+    }
+
+    // TODO: currently we sign them in. Need to resolve this!
+    if (!props.isUserLoggedIn) {
+        // TODO: sign in form
+        components = [<NewsLetterSignUps />, <OptOuts />];
```

Comment:
Yes, this is not _currently_ changing the components or their ordering, but id in place to facilitate future work.